trino-connector/src/main/java/com/datastrato/gravitino/trino/connector/catalog/CatalogConnectorManager.java

```diff
@@ -146,55 +149,75 @@ public void loadCatalogs(GravitinoMetalake metalake) {
     Set<String> catalogNameStrings =
         Arrays.stream(catalogNames).map(NameIdentifier::toString).collect(Collectors.toSet());
 
-    catalogConnectors.entrySet().stream()
-        .filter(
-            entry ->
-                !catalogNameStrings.contains(entry.getKey())
-                    &&
-                    // Skip the catalog doesn't belong to this metalake.
-                    entry.getValue().getMetalake().name().equals(metalake.name()))
-        .forEach(
-            (entry) -> {
-              catalogInjector.removeCatalogConnector(entry.getKey());
-              catalogConnectors.remove(entry.getKey());
-              LOG.info(
-                  "Remove catalog '{}' in metalake {} successfully.",
-                  entry.getKey(),
-                  metalake.name());
-            });
+    for (Map.Entry<String, CatalogConnectorContext> entry : catalogConnectors.entrySet()) {
+      if (!catalogNameStrings.contains(entry.getKey())
+          &&
+          // Skip the catalog doesn't belong to this metalake.
+          entry.getValue().getMetalake().name().equals(metalake.name())) {
+        unloadCatalog(metalake, entry.getKey());
+      }
+    }
 
     Arrays.stream(catalogNames)
         .forEach(
             (NameIdentifier nameIdentifier) -> {
-              String catalogName = "";
               try {
-                catalogName = nameIdentifier.toString();
-
-                // TODO (yuhui) need to handle catalog changed.
-                if (!catalogConnectors.containsKey(catalogName)) {
-                  Catalog catalog = metalake.loadCatalog(nameIdentifier);
-
+                Catalog catalog = metalake.loadCatalog(nameIdentifier);
+                GravitinoCatalog gravitinoCatalog = new GravitinoCatalog(metalake.name(), catalog);
+                if (!catalogConnectors.containsKey(gravitinoCatalog.getFullName())) {
+                  // Load new catalogs belows to the metalake.
                   if (catalog.type() == Catalog.Type.RELATIONAL) {
-                    GravitinoCatalog gravitinoCatalog =
-                        new GravitinoCatalog(metalake.name(), catalog);
-                    CatalogConnectorContext catalogConnectorContext =
-                        catalogConnectorFactory.loadCatalogConnector(metalake, gravitinoCatalog);
-
-                    catalogConnectors.put(catalogName, catalogConnectorContext);
-                    catalogInjector.injectCatalogConnector(catalogName);
-                    LOG.info(
-                        "Load catalog {} in metalake {} successfully.",
-                        catalogName,
-                        metalake.name());
+                    loadCatalog(metalake, gravitinoCatalog);
                   }
+                } else {
+                  // Reload catalogs that have been updated in Gravitino server.
+                  reloadCatalog(metalake, gravitinoCatalog);
```

Comment:
Based on your logic, will we only update the catalog when its name is changed? What if the name hasn't changed?

Reply:
Furthermore, what criteria are used to determine if a catalog has been altered or not?

Reply:
Not only is it, We keep the catalog information in the CatalogConnectorManager. The reload function compares the last modify time with the newly loaded catalog. if changed, do reloadCatalog.

Reply:
Let me see.